util/argo/resource_tracking.go

```diff
@@ -138,14 +141,25 @@ func UnstructuredToAppInstanceValue(un *unstructured.Unstructured, appName, name
 }
 
 // SetAppInstance set label/annotation base on tracking method
-func (rt *resourceTracking) SetAppInstance(un *unstructured.Unstructured, key, val, namespace string, trackingMethod v1alpha1.TrackingMethod) error {
+func (rt *resourceTracking) SetAppInstance(un *unstructured.Unstructured, key, parentAppName, parentAppNamespace string, trackingMethod v1alpha1.TrackingMethod) error {
```

Comment:
I feel like `parentAppName` and `parentAppNamespace` is too specific for app of apps pattern, I'd rather use `appName` and `appNamespace` to be more generic.